### PR TITLE
Reference app documentation: updates to top-level README (basic setup instructions)

### PR DESCRIPTION
**Pull request by aka-msft**
Adding most of the team so as to get input on whether there is anything missing they would like to see in this doc.

Trying to keep it simple yet accurate/robust enough for a first-time user to follow.

// TODO: Still need to fill out Docker section.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,27 +20,24 @@ contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additio
...
+2. Open a terminal in the local repo directory
+3. Run `npm install`
 
-### Transpile to JS
+### Build (Transpile to JS)
 
-```
-npm run build
-```
+1. `npm run build` will transpile source files from the `src/` directory into a destionation directory named `dist/`.
```
Comment (WilliamMortlMicrosoft) [inline README.md]: in sections like this you should explain what build does brielfy, then show a command line example, and what it should look like it it runs properly... this is done like:

```bash
$ amandas_awesome_command param1 param2
My super cool server wrote this text
{
   "value":1
   "value2": "and this awesome JSON"
}
```

^- those above ticks close the block

this is sort of the industry standard way to show users the command, while also giving them some comfort in showing them what the output should look like :-)

File: README.md [hunk 2]
```diff
@@ -20,27 +20,24 @@ contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additio
...
 
-### Run dev server
+### Test (Unit tests)
 
-```
-npm start
-```
+1. Create your unit test file adjacent to your source file in the `src/` directory.  Name your test as `*.test.ts`.
+2. From the project root, run `npm test` in the terminal.
```
Comment (WilliamMortlMicrosoft) [inline README.md]: same as above

File: README.md [hunk 3]
```diff
@@ -20,27 +20,24 @@ contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additio
...
+3. Output should show that your test ran.
 
-### Lint
+### Run dev server
 
-```
-npm run lint
-```
+1. `npm start` will start the server locally.  If no environment variable is provided, the application will run on the default port defined in `server.ts`.
```
Comment (WilliamMortlMicrosoft) [inline README.md]: same as above

File: README.md [hunk 4]
```diff
@@ -20,27 +20,36 @@ contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additio
 
 ### Install dependencies
 
-```
-npm install
-```
+1. Clone the repository
+2. Open a terminal in the local repo directory
+3. Run `npm install`
```
Comment (frodopwns) [inline README.md]: I like that you are adding more description. I'd prefer the code blocks stay blocks though. This way one can easily copy/paste the code therein.

File: README.md [hunk 5]
```diff
@@ -20,27 +20,36 @@ contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additio
...
 
-### Transpile to JS
+### Build (Transpile to JS)
+
+1. `npm run build` will transpile source files from the `src/` directory into a destionation directory named `dist/`.  Console output should appear as follows:
 
 ```
-npm run build
-```
+aka@MININT-81A1OI6 MINGW64 /c/users/aka/repos/helium (aka/root-readme-updates)
```
Comment (frodopwns) [inline README.md]: remove your personal information

File: README.md [hunk 6]
```diff
@@ -20,27 +20,36 @@ contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additio
...
+$ npm run build
 
-### Run dev server
+> helium@1.0.0 build C:\users\aka\repos\helium
```
Comment (frodopwns) [inline README.md]: personal username

File: README.md [hunk 7]
```diff
@@ -20,27 +20,36 @@ contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additio
...
+> gulp scripts
 
-```
-npm start
+[11:28:19] Using gulpfile C:\users\aka\repos\helium\gulpfile.js
```
Comment (frodopwns) [inline README.md]: personal username

File: README.md [hunk 8]
```diff
@@ -20,27 +20,36 @@ contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additio
...
+[11:28:19] Starting 'scripts'...
+[11:28:21] Finished 'scripts' after 1.97 s
 ```
 
-### Lint
+### Test (Unit tests)
 
-```
-npm run lint
-```
```
Comment (frodopwns) [inline README.md]: I prefer this style of code example because its easy to copy/paste

File: README.md (hunk 1 above)
Comment (frodopwns) [reply README.md]: I like it when it is easy to copy/paste the example commands. I know others prefer to make it harder to copy/paste. Let's address Readme cleanup after the lion's share of the important work is done.

**Pull request merged**